server/lib/backups3/__init__.py

```diff
@@ -0,0 +1,139 @@
+        except:
+            print("Failed to create bucket: %s" % bucket_name)
+
+
+    def list_all_buckets(self):
+        # Function to get all the buckets available in given endpoint url
+        try:
+            buckets = [bucket.name for bucket in self.s3.buckets.all()]
+            return buckets
+        except:
+            return []
+
+
+    def list_content_of_bucket(self, bucket_name):
+        # Function to get the content of a particular bucket
+        for bucket in self.s3.buckets.all():
+            if bucket.name == bucket_name:
+                try:
+                    contents = [file.key for file in bucket.objects.all()]
+                    return contents
+                except:
+                    return []
+
+
+    def download_data_from_bucket(self, bucket_name, filename, destination):
+        # Function to download object from given bucket.
+        dest = "%s/%s" % (destination, filename)
+        try:
+            self.s3.Bucket(bucket_name).download_file(filename, dest)
+            print("Downloaded: %s" % filename)
+        except:
+            print("Failed to Download: %s" % filename)
+
+
+    def download_whole_bucket(self, bucket_name, destination):
+        # Function to download the whole bucket content
+        try:
+            bucket_to_download = self.s3.Bucket(bucket_name)
+            print("Downloading contents of %s bucket:" % bucket_name)
+            for file in bucket_to_download.objects.all():
+                path, filename = os.path.split(file.key)
+                dest = "%s/%s" % (destination, filename)
+                bucket_to_download.download_file(file.key, dest)
+                print("Downloaded: %s" % filename)
+        except:
+            print("Failed to Download: %s" % bucket_name)
+
+
+    def delete_from_bucket(self, bucket_name):
+        # Function to delete from bucket
+        try:
+            print("Deleting from %s bucket:" % bucket_name)
+            for bucket in self.s3.buckets.all():
+                if bucket.name == bucket_name:
+                    for file in bucket.objects.all():
+                        path, filename = os.path.split(file.key)
+                        file.delete()
+                        print("Deleted: %s" % filename)
+        except:
+            print("Failed to delete objects from: %s" % bucket_name)
+
+
+    def delete_the_bucket(self, bucket_name):
+        # Function to delete the bcuket itself
+        try:
+            for bucket in self.s3.buckets.all():
+                if bucket.name == bucket_name:
+                    print("Bucket: %s need to be empty before it can be deleted." %
+￼                   bucket_name)
+                    delete_from_bucket(bucket_name)
+                    bucket.delete()
+                    print("Deleted bucket: %s" % bucket_name)
+        except:
+            print("Failed to delete bucket: %s" % bucket_name)
+
+
+    def upload_data_to_bucket(self):
```

Comment:
This knows too much. Let the caller loop over the archive: this method should know nothing about that. It should take a tarball as argument and store it as a single object, so only lines 121-137 should be included (with appropriate modifications) in this method.